### PR TITLE
Use DEFAULT_TO_CXX to control both linker and compiler

### PR DESCRIPTION
**Pull request by sbc100**
See https://github.com/emscripten-core/emscripten/issues/11121

### CODE REVIEW

File: tests/common.py [hunk 1]
```diff
@@ -553,9 +553,10 @@ def build(self, filename, libraries=[], includes=[], force_c=False, js_outfile=T
     compiler = [compiler_for(filename, force_c)]
     if compiler[0] == EMCC:
       # TODO(https://github.com/emscripten-core/emscripten/issues/11121)
-      # We link with C++ stdlibs, even when linking with emcc for historical reasons.  We can remove
-      # this if this issues is fixed.
-      compiler.append('-nostdlib++')
+      # For historical reasons emcc compiles and link as C++ by default.
```
Comment (kripken) [inline tests/common.py:556]:
```suggestion
      # For historical reasons emcc compiles and links as C++ by default.
```

File: tests/common.py [hunk 2]
```diff
@@ -553,9 +553,10 @@ def build(self, filename, libraries=[], includes=[], force_c=False, js_outfile=T
...
+      # However we want to run out tests in a more strict manor.  We can
```
Comment (kripken) [inline tests/common.py:557]:
```suggestion
      # However we want to run our tests in a more strict manner.  We can
```

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
Test failure on `test_pthread_nested_work_queue` looks random? But I don't recognize that test name...

**Comment by sbc100** [comment]
Yup, that looks like a flake although I don't remember seening that one flake before.

**Comment by kripken** [comment]
Ah, I just saw it again (on an unopen PR) so it looks like a very recent regression, maybe not a flake.

**Comment by sbc100** [comment]
This change wouldn't have landed if it wasn't a flake.. but maybe its new flake.

**Pull request merged**